src/utils/KernelVaultViewer.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {
+    AccessControlUpgradeable,
+    Address,
+    ERC20PermitUpgradeable,
+    ERC20Upgradeable,
+    IERC20,
+    IERC20Metadata,
+    Math,
+    ReentrancyGuardUpgradeable,
+    SafeERC20
+} from "lib/yieldnest-vault/src/Common.sol";
+
+import {KernelStrategy} from "src/KernelStrategy.sol";
+import {IProvider} from "src/interface/IProvider.sol";
+import {IStrategy} from "src/interface/IStrategy.sol";
+import {IVault} from "src/interface/IVault.sol";
+
+import {IStakerGateway} from "src/interface/external/kernel/IStakerGateway.sol";
+import {Guard} from "src/module/Guard.sol";
+
+import {BaseVaultViewer} from "lib/yieldnest-vault/src/utils/BaseVaultViewer.sol";
+
+contract KernelVaultViewer is BaseVaultViewer {
+    using Math for uint256;
+
+    function vault() public view returns (KernelStrategy) {
+        return KernelStrategy(payable(address(_getStorage().vault)));
+    }
+    /**
+     * @notice Returns the maximum amount of assets that can be withdrawn for a specific asset by a given owner.
+     * @param asset_ The address of the asset.
+     * @param owner The address of the owner.
+     * @return maxAssets The maximum amount of assets.
+     */
+
+    function maxWithdrawAsset(address asset_, address owner) public view returns (uint256 maxAssets) {
```

Comment:
This will return incorrectly for the clis strategy since it handles deposits and withdrawals differently @danoctavian 
Do you want to add another version of this KernelVaultViewer specific to the clis strategy ?

Reply:
Yes let's do that, maybe inheriting this to reuse some of the functions below?

Reply:
added in https://github.com/yieldnest/yieldnest-kernel-lrt/pull/27